catkin_ws/src/planner/src/substates/navigate_buoy.py

```diff
@@ -47,23 +47,14 @@ def execute(self, ud):
 
         self.control.rotateEuler(homing_rotation) # bring to exact angle 
         self.control.move(homing_position) # move in front of buoy
-        print("Successfully centered in front of gate")
+        print("Successfully centered in front of buoy")      
 
-        buoy_symbol_locations = 12 if buoy_object[5] is None else buoy_object[5]
-        first_buoy_symbol_location = math.floor(buoy_symbol_locations / 10)
-        second_buoy_symbol_location = buoy_symbol_locations - 10 * math.floor(buoy_symbol_locations / 10)
-
-        for symbol_location in [first_buoy_symbol_location, second_buoy_symbol_location]:
+        symbol_objects = self.mapping.getClass(cls=self.target_symbol)
+        
+        for symbol in symbol_objects:
+            self.control.move((symbol[1], symbol[2], symbol[3])) # move to symbol
             self.control.move(homing_position) # move in front of buoy
-            if symbol_location == 1: # TOP LEFT
-                self.control.moveDeltaLocal(offset_distance, self.buoy_width/4, self.buoy_height/4)
-            elif symbol_location == 2: # TOP RIGHT
-                self.control.moveDeltaLocal(offset_distance, -self.buoy_width/4, self.buoy_height/4)
-            elif symbol_location == 3: # BOTTOM LEFT
-                self.control.moveDeltaLocal(offset_distance, self.buoy_width/4, -self.buoy_height/4)
-            elif symbol_location == 4: # BOTTOM RIGHT
-                self.control.moveDeltaLocal(offset_distance, -self.buoy_width/4, -self.buoy_height/4)
-
+            self.control.rotateEuler(homing_rotation) # bring to exact angle 
```

Comment:
Very skeptical this will work. At the very least insert 0s into homing rotation instead of Nones

Reply:
In general this file has a lot of steps that aren't clear why they are there. For example the offset. Is that to put the AUV not too close?

Reply:
Yeah true, it could be cleaner for sure. I'll replace the None's with zeroes instead, but for the overall structure I feel like that's not a regression as much as something that was already a problem. We could give new members the task of cleaning up some of these missions/commenting them more thoroughly? Or I could do it now but I feel like lots of the missions are in a similar state

Reply:
we could let the new members take care of this. It's mostly a copy-paste task with a little bit of ros formatting. It's good for making them familiar with how the planner works

Reply:
Sounds good, I'll merge this in and add a task to the board for cleaning up missions/commenting